syntax = "proto3";
package user;
import "google/protobuf/empty.proto";

enum SexType {
  man = 0;
  women = 1;
}

// create user
message CreateUserRequest {
  string uid = 1;
  string user_name = 2;
  string password = 3;
  SexType sex = 4;
}

// delete user
message DeleteUserRequest {
  string uid = 1;
}

// login user
message LoginUserRequest {
  string uid = 1;
  string password = 2;
}
message LoginUserResult {
  string uid = 1;
  string user_name = 2;
  string token = 3;
}

// logout user
message LogoutUserRequest {
  string uid = 1;
  string token = 2;
}

message GetUidByTokenRequest {
  string token = 1;
}
message GetUidByTokenResult {
  string uid = 1;
}

// pait: {"group": "user", "tag": [["grpc-user", "grpc_user_service"]]}
service User {
  // The interface should not be exposed for external use
  // pait: {"enable": false}
  rpc get_uid_by_token (GetUidByTokenRequest) returns (GetUidByTokenResult);
  // pait: {"summary": "User exit from the system", "url": "/user/logout"}
  rpc logout_user (LogoutUserRequest) returns (google.protobuf.Empty);
  // pait: {"summary": "User login to system", "url": "/user/login"}
  rpc login_user(LoginUserRequest) returns (LoginUserResult);
  // pait: {"tag": [["grpc-user", "grpc_user_service"], ["grpc-user-system", "grpc_user_service"]]}
  // pait: {"summary": "Create users through the system", "url": "/user/create"}
  rpc create_user(CreateUserRequest) returns (google.protobuf.Empty);
  // pait: {"url": "/user/delete", "tag": [["grpc-user", "grpc_user_service"], ["grpc-user-system", "grpc_user_service"]]}
  // pait: {"desc": "This interface performs a logical delete, not a physical delete"}
  rpc delete_user(DeleteUserRequest) returns (google.protobuf.Empty);
}
